lib/rules/headers/h2-toc.js

```diff
@@ -5,18 +5,35 @@ exports.name  = 'headers.h2-toc';
 
 exports.check = function (sr, done) {
 
-    var EXPECTED_HEADING = /Table\ of\ Contents/;
-
-    var toc = sr.$('h2')
-    ,   found = false;
-
-    toc.each(function() {
-        found = found || EXPECTED_HEADING.test(sr.$(this).text());
-    });
-
-    if (!found) sr.warning(exports.name, 'missing');
-
-    return done();
+    var EXPECTED_HEADING = /table\s+of\s+contents/i;
+    var toc = sr.$('nav#toc > h2');
+
+    if (1 === toc.length) {
+        if (!EXPECTED_HEADING.test(sr.$(toc[0]).text())) {
+            sr.error(exports.name, 'not-found');
+        }
+    }
+    else {
+        toc = sr.$('#toc > h2');
```

Comment:
From the discussion with @plehegar and Fantasai, we said we would either accept `<nav>` or `<div>`. I don't think we should allow other tags.